src/Microsoft.ML.Core/Data/ColumnTypeExtensions.cs

```diff
@@ -17,5 +17,10 @@ internal static class ColumnTypeExtensions
         public static bool IsStandardScalar(this ColumnType columnType) =>
             (columnType is NumberType) || (columnType is TextType) || (columnType is BoolType) ||
             (columnType is TimeSpanType) || (columnType is DateTimeType) || (columnType is DateTimeOffsetType);
+
+        /// <summary>
+        /// Zero return means either it's not a key type or the cardinality is unknown.
+        /// </summary>
+        public static int KeyCount(this ColumnType columnType) => (columnType as KeyType)?.Count ?? 0;
     }
```

Comment:
Why do we keep this extension? You very nicely cast to KeyType and check inline with (type is KeyType key) and then access the Count field directly. Isn't that a more correct access the Count field?

Reply:
It's just a convenience/helper for the cases where we would have cast to KeyType anyway. This just consolidates the code, so we don't have to have the `as KeyType` and `?? 0` everywhere. This line of code would need to be duplicated 59 times if we didn't keep it.